### ISSUE TITLE
Backend accepts messages posted from origins besides localhost.

### ISSUE THREAD

**Issue by KodyJKing** [wailsapp/wails#3162]
### Description

On the WebView2 frontend, if a Wails app navigates to an external site, that site can call any exposed backend APIs. This seems like a social engineering risk if an attacker finds any sneaky way to get a user to navigate away from localhost.

### To Reproduce

1. Create a Wails app.
2. Link to this page https://kodyjking.github.io/wails-xss/ (source: https://github.com/KodyJKing/wails-xss/blob/main/index.html)
3. Follow the link from your wails app.

### Expected behaviour

The backend ignores messages posted from this origin.

### System Details

```shell
# Wails
Version | v2.6.0

# System
┌────────────────────────────────────┐
| OS           | Windows 10 Home     |
| Version      | 2009 (Build: 19045) |
| ID           | 22H2                |
| Go Version   | go1.21.3            |
| Platform     | windows             |
| Architecture | amd64               |
└────────────────────────────────────┘

# Dependencies
┌───────────────────────────────────────────────────────┐
| Dependency | Package Name | Status    | Version       |
| WebView2   | N/A          | Installed | 120.0.2210.91 |
| Nodejs     | N/A          | Installed | 20.9.0        |
| npm        | N/A          | Installed | 10.1.0        |
| *upx       | N/A          | Available |               |
| *nsis      | N/A          | Available |               |
└─────────────── * - Optional Dependency ───────────────┘
```

**Comment by leaanthony** [comment]
Certainly something we will look at though we don't really recommend navigating to untrusted sites in your desktop apps!

**Comment by KodyJKing** [comment]
Makes sense. My app *should* stay on localhost, but I'm anxious about my attack surface including any library that can affect navigation.

**Comment by leaanthony** [comment]
Yeah for sure, but there would need to be quite a lot of knowledge on the internal calls to construct something that didn't error. We will of course look at ways to mitigate this.

**Comment by leaanthony** [comment]
Hey there\!

First off, thank you for taking the time to open this issue. Community feedback like yours has been instrumental in shaping Wails into what it is today.

We're doing some housekeeping on our issue tracker as we focus on the v3 release. This issue has been open for a while, and honestly, with the project's growth and the sheer volume of feedback we receive, some things inevitably slip through the cracks. That's on us, not you.

We're closing this issue, but that doesn't mean it wasn't valuable. If this is still relevant:

- For bugs: Please test with the latest version (or v3 alpha) and open a fresh issue if it persists
- For features: Feel free to open a new issue with updated context

Thank you for being part of the Wails journey. Your patience and support mean more than you know.
